roxie/roxiemem/roxiemem.hpp

```diff
@@ -199,21 +199,24 @@ class roxiemem_decl DataBuffer
     void released();
 
 protected:
-    DataBuffer()
+    DataBuffer() : count(1)
     {
-        atomic_set(&count,1);  // Starts off active
     }
 public:
-    void Link() { atomic_inc(&count); }
+    // Link and release are used to keep count of the references to the buffers.
+    void Link()
+    {
+        count.fetch_add(1, std::memory_order_relaxed);
+    }
     void Release();
     inline unsigned queryCount() const
     {
-        return atomic_read(&count);
+        return count.load(std::memory_order_relaxed);
```

Comment:
Yes, because doesn't need coordination with any other fields.  It should really be replaced with isEmpty() - since it is only valid to compare it with 1.